src/Public/OpenApi.ps1

```diff
@@ -583,9 +753,102 @@ function Add-PodeOAComponentSchema {
         [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
         [hashtable]
         $Schema
+
     )
 
     $PodeContext.Server.OpenAPI.components.schemas[$Name] = ($Schema | ConvertTo-PodeOASchemaProperty)
+
+    $json = $PodeContext.Server.OpenAPI.components.schemas[$Name] | ConvertTo-Json -Depth 20 -Compress
+    $obj = ConvertFrom-Json $json -AsHashtable
+    Resolve-References -obj $obj -schemas $PodeContext.Server.OpenAPI.components.schemas
+
+    $PodeContext.Server.OpenAPI.hiddenComponents.schemaJson[$Name] = $obj | ConvertTo-Json -Depth 20
```

Comment:
Is this section absolutely mandatory? From what I can tell its only really done so that `Test-PodeOARequestSchema` works if optionally invoked? If it needs to stay then:

* The `Resolve-References` doesn't meet the naming standard, and should be `Resolve-PodeOAReferences` instead. The parameter block in this function also needs to be a `param(..)` block rather than inline with the function.
* The `-AsHashtable` parameter on `ConvertFrom-Json` doesn't exist in PS5.1 - which could mean this is PS7+ only.
* In the `Test-PodeOARequestSchema` function, `Test-Json` is also not in PS5.1.
* Other than a PS7 check, if this is completely optional I think a `-Resolve` switch on this function would be good - not it's not forced every time.

Reply:
Revisited the function and removed redundant code